src/http/controllers/health.rs

```diff
@@ -1,39 +1,31 @@
 use crate::config::Config;
-use saphir::{Method, *};
+use saphir::{
+    controller::Controller,
+    http::{Method, StatusCode},
+    macros::controller,
+};
 use std::sync::Arc;
 
-struct ControllerData {
-    config: Arc<Config>,
-}
-
 pub struct HealthController {
-    dispatch: ControllerDispatch<ControllerData>,
+    config: Arc<Config>,
 }
 
 impl HealthController {
     pub fn new(config: Arc<Config>) -> Self {
-        let dispatch = ControllerDispatch::new(ControllerData { config });
-        dispatch.add(Method::GET, "/", health);
-
-        HealthController { dispatch }
+        Self { config }
     }
 }
 
-impl Controller for HealthController {
-    fn handle(&self, req: &mut SyncRequest, res: &mut SyncResponse) {
-        self.dispatch.dispatch(req, res);
-    }
-
-    fn base_path(&self) -> &str {
-        "/health"
+#[controller(name = "health")]
+impl HealthController {
+    #[get("/")]
+    async fn get_health(&self) -> (u16, String) {
+        build_health_response(&self.config)
     }
 }
 
-fn health(controller: &ControllerData, _req: &SyncRequest, res: &mut SyncResponse) {
-    build_health_response(res, &controller.config.hostname);
-}
-
-pub fn build_health_response(res: &mut SyncResponse, hostname: &str) {
-    res.status(StatusCode::OK)
-        .body(format!("Devolutions Gateway \"{}\" is alive and healthy.", hostname));
+pub fn build_health_response(config: &Config) -> (u16, String) {
```

Comment:
I think you could return `StatusCode` directly instead of `u16`
Doing so you can remove all the `as_u16` in all handlers

Reply:
Thanks, I didn't know that it is possible to use StatusCode directly -- in saphir docs examples there is the only example with u16 return code type :smile:

Reply:
Haha maybe doc could be improved to discover this more easily: you can actually return anything implementing their `Responder` trait, including:
https://docs.rs/saphir/2.8.0/saphir/responder/trait.Responder.html#foreign-impls